main.tf

```diff
@@ -1,10 +1,17 @@
-provider "azurerm" {
-    # The "feature" block is required for AzureRM provider 2.x. 
-    # If you are using version 1.x, the "features" block is not allowed.
-    version = "~>2.0"
-    features {}
+terraform {
```

Comment:
Maybe move the `required_providers` into the `terraform` block defined at the end of the file.